src/cli/transaction.go

```diff
@@ -74,6 +76,50 @@ func decodeRawTxnCmd() *cobra.Command {
 	}
 }
 
+func encodeJSONTxnCmd() *cobra.Command {
+	return &cobra.Command{
+		Short:                 "Encode JSON transaction",
+		Use:                   "encodeJsonTransaction [file path or -]",
+		DisableFlagsInUseLine: true,
+		SilenceUsage:          true,
+		Args:                  cobra.ExactArgs(1),
+		RunE: func(_ *cobra.Command, args []string) error {
+			jsonFilePath := args[0]
+			var jsonFile *os.File
+			var err error
+			if jsonFilePath == "-" {
+				jsonFile = os.Stdin
+				err = nil
+				jsonFilePath = "<stdin>"
+			} else {
+				jsonFile, err = os.Open(jsonFilePath)
+			}
+			if err != nil {
+				return fmt.Errorf("open file failed %s:\n %v", jsonFilePath, err)
+			}
+			var rTxn readable.Transaction
+			err = json.NewDecoder(jsonFile).Decode(&rTxn)
+			if err != nil {
+				return fmt.Errorf("invalid JSON transaction: %v", err)
+			}
+
+			txn, err := rTxn.GetObject()
+			if err != nil {
+				return err
+			}
+			rawTxn, err := txn.SerializeHex()
+			if err != nil {
+				return err
+			}
+			return printJSON(struct {
```

Comment:
@gz-c `createTransaction` command also returns raw transaction . However it outputs hex string rather than JSON . Shall we do the same in this case ?

Reply:
`createTransaction` returns encoded raw transaction, because that is the input necessary for `injectTransaction`

Reply:
Done ! Guess this is ready for review .